### ISSUE TITLE
Important parts of error message are barely readable when light terminal theme is used

### ISSUE THREAD

**Issue by michallepicki** [roc-lang/roc#4136]
Hi! 👋🏻 This is very tiny issue, but I didn't find any discussions about it in the issues tracker. I can work-around it easily by switching to a different terminal theme temporarily.

I am using the "Solarized Light" theme for my terminal, and because parts of the error message text has a white-ish color, this text has too little contrast compared to the background to be readable easily.
![image](https://user-images.githubusercontent.com/4972756/193450626-f0533a02-0048-40fb-9418-5637a277e8c6.png)

I get similar effect with "Tango Light" and "GNOME Light" themes

**Comment by rtfeldman** [comment]
Yikes! That is indeed not readable. 😅

**Comment by mkrieger1** [comment]
The problem seems to be that the palette that `default_palette_from_style_codes` (in `crates/reporting/src/report.rs`) returns uses "white" (for `primary` and `code_block`). I think the "black" and "white" ANSI colors should not be used here, because they do not take into account whether a light or dark colorscheme is used in the terminal. Presumably the default foreground text color should be used instead.

**Comment by rtfeldman** [comment]
Good point! It seems like the fix here is to use the default terminal color for text (so, don't use any ANSI escapes on it), rather than hardcoding it to white.

@brian-carroll would that break anything for the web repl colors?

**Comment by brian-carroll** [comment]
It sounds like it would work fine.
Should be tested. Might need some CSS changes.
